mava/utils/done_future.py

```diff
@@ -0,0 +1,41 @@
+from concurrent import futures
+from typing import Any, Callable, Generic, Optional, TypeVar
+
+T = TypeVar("T")  # type of result
+
+
+class DoneFuture(futures.Future, Generic[T]):
```

Comment:
This looks really nice :raised_hands: Thanks @sash-a :slightly_smiling_face:

Reply:
Thanks Dries :smile: